Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/ExposureKeyHistoryCalculations.kt

```diff
@@ -1,39 +1,70 @@
 package de.rki.coronawarnapp.submission
 
+import KeyExportFormat
+import androidx.annotation.VisibleForTesting
 import com.google.android.gms.nearby.exposurenotification.TemporaryExposureKey
 
 class ExposureKeyHistoryCalculations(
+    private val submissionStatusRepository: SubmissionStatusRepository,
     private val transmissionRiskVectorDeterminator: TransmissionRiskVectorDeterminator,
+    private val daysSinceOnsetOfSymptomsVectorDeterminator: DaysSinceOnsetOfSymptomsVectorDeterminator,
     private val keyConverter: KeyConverter
 ) {
 
+    companion object {
+        const val VECTOR_LENGTH = 15
+    }
+
     fun transformToKeyHistoryInExternalFormat(
         keys: List<TemporaryExposureKey>,
         symptoms: Symptoms
     ) =
         toExternalFormat(
-            toSortedHistory(limitKeyCount(keys)),
-            transmissionRiskVectorDeterminator.determine(symptoms)
+            limitKeyCount(toSortedHistory(keys)),
+            transmissionRiskVectorDeterminator.determine(symptoms),
+            daysSinceOnsetOfSymptomsVectorDeterminator.determine(symptoms, VECTOR_LENGTH)
         )
 
-    fun <T> limitKeyCount(keys: List<T>): List<T> =
-        keys.take(MAXIMUM_KEYS)
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun limitKeyCount(keys: List<TemporaryExposureKey>) =
+        keys.filter { it.daysAgo <= 14 }
 
-    fun toExternalFormat(
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun toExternalFormat(
         keys: List<TemporaryExposureKey>,
-        transmissionRiskVector: TransmissionRiskVector
-    ) =
-        keys.mapIndexed { index, key ->
-            // The latest key we receive is from yesterday (i.e. 1 day ago),
-            // thus we need use index+1
-            keyConverter.toExternalFormat(key, transmissionRiskVector.getRiskValue(index + 1))
+        transmissionRiskVector: TransmissionRiskVector,
+        daysSinceOnsetOfSymptomsVector: DaysSinceOnsetOfSymptomsVector
+    ): List<KeyExportFormat.TemporaryExposureKey> {
+        val result = mutableListOf<KeyExportFormat.TemporaryExposureKey>()
+        keys.groupBy { it.daysAgo }.forEach { entry ->
+            val index = daysSinceOnsetOfSymptomsVector.indexOf(entry.key)
```

Comment:
Not sure if this is correct. If `entry.key` is semantically `daysAgo` (i.e. the age of the TEK in days), might work partially if the vector is [0;14] but even with a date, we can also have [-14;0] and with a range onset of symptoms, the values in the vector are in the 700s, 2000s, 3000s, or 4000s.